sklearn/ensemble/_hist_gradient_boosting/_predictor.pyx

```diff
@@ -14,7 +14,7 @@ from ._bitset cimport in_bitset_2d_memoryview
 
 
 def _predict_from_raw_data(  # raw data = non-binned data
-        node_struct [:] nodes,
+        const node_struct [:] nodes,
```

Comment:
I think this better be introduced in another PR.

Reply:
The reason I introduced this in the same PR is because the tests which fail are present in this PR. So checking that the changes work correctly is easier to track here.

Reply:
@jjerphan Could you kindly review the PR whenever you have the time?

Reply:
Can you revert the changes made to this file and instead define `BaseHistGradientBoosting._more_tags`?

Reply:
Yes, I think this better be done in another PR as hinted by https://github.com/scikit-learn/scikit-learn/pull/25624#issuecomment-1433354801.

Generally, we tend to have the scope of PR atomic, because it is easier tend to search for potential regression a PR might have caused (basically this allows avoiding not git-bisecting and makes the changelog exact).

Reply:
Okay, understood.

Reply:
The rule of thumb generally is: 1 PR, 1 self-contained/atomic change.

This is rather conservatist as a choice but this also better stability.